### PR TITLE
Rewrite api requests

### PR DESCRIPTION
**Pull request by GusCaplan**
testing was done on googlebot for 24 hours, with no issues in uptime or increase in http errors

request speed was tested with 30 requests to `https://www.google.com/` and saw a 10-40ms speedup

memory usage: uses 16.7% memory that superagent does.

### CODE REVIEW

**Comment by Gawdl3y** [comment]
Stop putting unrelated changes in your PRs, or get shanked by the :dagger: of lordliness.

File: src/util/Request/FormData.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class FormData {
+  constructor() {
+    this.boundary = '--dicksword.js--';
```
Comment (hydrabolt) [inline src/util/Request/FormData.js]: are you srs no

File: src/util/Request/Request.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+          response.status = response.statusCode;
+          if (response.statusCode >= 400) {
+            reject(response);
+          } else {
+            resolve(response);
+          }
+        });
+      };
+      let request;
```
Comment (hydrabolt) [inline src/util/Request/Request.js]: you can ternary this and make it a const

**Comment by GusCaplan** [comment]
fuck memory usage and request speed and http and life

**Pull request closed without merging**